### DISCUSSION TITLE
PMM Enforcement of .NFO Metadata

### DISCUSSION THREAD

**Discussion by zpz5HAU-tgc3fgw2xwr** [meisnate12/Plex-Meta-Manager#357] [Ideas]
I'm surprised I wasn't able to find any discussions or issues about this already, but I've been searching around the last few days and found an interesting custom Plex Agent:

https://github.com/gboudreau/XBMCnfoTVImporter.bundle
https://github.com/gboudreau/XBMCnfoMoviesImporter.bundle

I haven't tested these myself, because I'm personally not a fan of Plex Plug-Ins or custom agents since Plex has discontinued support. But the comments I've read say they work well.

Because I'm currently using PMM and like the way it's been working, I figured I'd bring this idea over here first before messing up my server with .bundle files.

Basically this would be an option to have PMM browse for matching NFO files wherever the video file is, and it would use this NFO file to override any relevant metadata. And because NFO files are standard for this stuff, this would also be a good way to clean up any really long Shows.yml or Movies.yml files people have just for metadata overrides.

**Comment by meisnate12** [comment]
wont the agent pick up that metadata? if so there's no real need for PMM to do it

**Reply by zpz5HAU-tgc3fgw2xwr** [reply]
Plex doesn't support NFO files at all (which has never made sense to me)

The agents I linked might work for now, but because they aren't supported who knows how long that'll actually be a solution (assuming it works still).

I've also read comments that the two bundles I linked don't play nice with the new Beta TV/Movie agents Plex has, but I can't confirm

**Reply by zpz5HAU-tgc3fgw2xwr** [reply]
And by "because they aren't supported" I mean Plex has officially said they don't support these plugins and agents and are phasing them out eventually. Not that the developer of that agent doesn't support it.

**Reply by meisnate12** [reply]
I'm sure those agents support NFO files because the entire function of an Agent is to obtain the metadata for items found in Plex. The NFO agents I assume support NFO files in the same directory. 

Either way, what you're asking for is at the end of the day an NFO agent which if I did make this functionality id probably just make it as an agent as they have or just edit theirs so I wouldn't have to start from scratch. 

P.S. One thing to note about PMM is that it doesn't actually have or need access to the file system where your video files are stored. So to check around the files would be a huge change in code.

**Reply by zpz5HAU-tgc3fgw2xwr** [reply]
Right, but I'm talking about future-proofing. If you went ahead and made an agent instead of an extension of PMM, then your agent would stop working when their agents stop working. Whereas PMM will work independent of Plex permanently breaking custom bundles.

I also figured because PMM has the filepath filter, that would provide a starting point for something like this. But you would know better than me on that subject.

**Reply by meisnate12** [reply]
I don't think Plex said they were going to end the ability to use 3rd party agents and Scanners I thought it was just plug-ins.

And I get the filepath from Plex as part of the metadata but I don't actually see the file system at all.

**Reply by zpz5HAU-tgc3fgw2xwr** [reply]
Hm... Well if that's the case I can still see a way for it to work by creating a folder of nfo's and naming the nfo file the same way you match up posters to tv shows. But if we go ahead and do that you lose the organization and cross-platform (ex: EMBY) benefits of keeping the NFO next to the video file, so you might as well not be using NFOs at all.


Well what about this then: **What if we could optionally split up the shows listed under metadata into their own YML files?**

This separation would help me with a few shows that are currently using ~120-300 lines (and growing) in my Shows.YML. There are also some overrides I want to do but avoid because I know it would take up way too much space in Shows.YML.

This would also make sharing these configs a little cleaner, just drag and drop a file instead of searching and cutting your selection

**Reply by meisnate12** [reply]
PMM can already have multiple metadata files btw you can split it up however you want

**Reply by zpz5HAU-tgc3fgw2xwr** [reply]
Oh, shit... Is that new? I don't remember seeing this on the wiki when I was setting everything up way back when

Judging by the multiple git examples, you'd format it like this?
```
libraries:
  Movies:
    metadata_path:
       - file: file1.yml
       - file: file2.yml
```

**Reply by zpz5HAU-tgc3fgw2xwr** [reply]
Any way to point it to a folder of files instead of individual files?

**Reply by meisnate12** [reply]
no but i could add folder which will grab every yml file in said folder

**Reply by zpz5HAU-tgc3fgw2xwr** [reply]
That would be awesome - thanks
